### DISCUSSION TITLE
Accessing property inside a callback with Auto generate selectors

### DISCUSSION THREAD

**Discussion by phoenixcz** [pmndrs/zustand#2250] [Q&A] [answered]
Hi,
I noticed that when the store is created via [Auto generate selectors](https://github.com/pmndrs/zustand/blob/main/docs/guides/auto-generating-selectors.md) my IDE doesn't display _React Hook cannot be called inside a callback_ error.

Does this mean I can access property inside a callback or IDE just doesn't have proper context to display this error?
![image](https://github.com/pmndrs/zustand/assets/18397435/21853905-52cd-4847-a1cc-39f2fe89b01b)

**Comment by charkour** [comment]
Hi @phoenixcz,

This ESLint warning is due to calling a hook inside of a `useEffect` callback which violates the rules of hooks: https://react.dev/warnings/invalid-hook-call-warning

Please try moving your hook calls outside of the `useEffect`.

**Reply by phoenixcz** [reply]
Hi @charkour,

Thanks for your answer.

I'm actually wondering about the other line. When I access store property via auto generated selector, I don't get any errors. But I'm not sure if it is safe to access the property in this way. 

![image](https://github.com/pmndrs/zustand/assets/18397435/ac376f66-6342-4ad7-986d-fe840de7cb2d)

**Reply by charkour** [reply, marked as answer]
Hi @phoenixcz. Good question! Even though there are no ESLint errors, it's not safe to use the store because you're calling the hook inside a callback which violates the rules of hooks.

**Reply by phoenixcz** [reply]
Thanks for your answer @charkour. 
Have a great day!
